### PR TITLE
Bug #128957 fix: Add single dashboard menu item >> incorrect link shown in menu

### PR DESCRIPTION
**Pull request by monikaatvowel**

### CODE REVIEW

File: src/components/com_tjdashboard/administrator/languages/en-GB/en-GB.com_tjdashboard.sys.ini [hunk 1]
```diff
@@ -1,9 +1,11 @@
 COM_TJDASHBOARD="TjDashboard"
 COM_TJDASHBOARD_XML_DESCRIPTION=""
-COM_TJDASHBOARD_TITLE_DASHBOARDS="Dashboards"
+COM_TJDASHBOARD_TITLE_DASHBOARDS="Dashboards"
 ;Added By Sudhir
```
Comment (twsvaishali) [inline src/components/com_tjdashboard/administrator/languages/en-GB/en-GB.com_tjdashboard.sys.ini]: @monikaatvowel  Please remove the block comments like this

File: src/components/com_tjdashboard/administrator/models/dashboard.php [hunk 2]
```diff
@@ -97,14 +97,43 @@ public function save($data)
 			return false;
 		}
 
+		$result = $dashboard->save();
+
 		// Store the data.
-		if (!$dashboard->save())
```
Comment (twsvaishali) [inline src/components/com_tjdashboard/administrator/models/dashboard.php]: @monikaatvowel  nothing special is done in this save method. Please check if you can remove this

File: src/components/com_tjdashboard/administrator/models/dashboard.php [hunk 3]
```diff
@@ -97,14 +97,43 @@ public function save($data)
...
+
+	/**
+	 * Method to get dashboard data.
+	 *
+	 * @param   integer  $itemId  The id of the dashboard.
+	 *
+	 * @return  mixed  Content item data object on success, false on failure.
+	 */
+
+	public function getItem($itemId = null)
```
Comment (twsvaishali) [inline src/components/com_tjdashboard/administrator/models/dashboard.php]: @monikaatvowel Same with this function.

File: src/components/com_tjdashboard/administrator/models/dashboard.php [hunk 4]
```diff
@@ -97,14 +97,35 @@ public function save($data)
...
+	}
+
+	/**
+	 * Method to auto-populate the model state.
+	 *
+	 * Note. Calling getState in this method will result in recursion.
+	 *
+	 * @return   void
+	 *
+	 * @since    1.6
```
Comment (thite-amol) [inline src/components/com_tjdashboard/administrator/models/dashboard.php]: @monikaatvowel Take care of versions

File: src/components/com_tjdashboard/administrator/models/dashboard.php [hunk 5]
```diff
@@ -97,14 +97,35 @@ public function save($data)
...
-		if (!$dashboard->save())
+		if (!$result)
 		{
 			$this->setError($dashboard->getError());
 
 			return false;
 		}
 
-		return $dashboard->dashboard_id;
+		$this->setState('dashboard.dashboard_id', $result);
```
Comment (thite-amol) [inline src/components/com_tjdashboard/administrator/models/dashboard.php]: @monikaatvowel you don't require both the statement (If you are using state then don't return the value)

File: src/components/com_tjdashboard/administrator/models/fields/dashboard.php [hunk 6]
```diff
@@ -0,0 +1,177 @@
+<?php
+/**
+ * @package     Joomla.Administrator
+ * @subpackage  com_contact
```
Comment (thite-amol) [inline src/components/com_tjdashboard/administrator/models/fields/dashboard.php]: @monikaatvowel Update the doc comment

File: src/components/com_tjdashboard/administrator/models/fields/dashboard.php [hunk 7]
```diff
@@ -0,0 +1,177 @@
...
+ */
+
+defined('JPATH_BASE') or die;
+
+/**
+ * Supports a modal contact picker.
+ *
+ * @since  1.6
+ */
+class JFormFieldModal_Dashboard extends JFormField
```
Comment (thite-amol) [inline src/components/com_tjdashboard/administrator/models/fields/dashboard.php]: @monikaatvowel Use camel case

File: src/components/com_tjdashboard/administrator/models/fields/dashboard.php [hunk 8]
```diff
@@ -0,0 +1,177 @@
...
+ * @subpackage  com_contact
+ *
+ * @copyright   Copyright (C) 2005 - 2018 Open Source Matters, Inc. All rights reserved.
+ * @license     GNU General Public License version 2 or later; see LICENSE.txt
+ */
+
+defined('JPATH_BASE') or die;
+
+/**
+ * Supports a modal contact picker.
```
Comment (thite-amol) [inline src/components/com_tjdashboard/administrator/models/fields/dashboard.php]: @monikaatvowel Update the purpose of this class here

File: src/components/com_tjdashboard/administrator/models/fields/modaldashboard.php [hunk 9]
```diff
@@ -0,0 +1,164 @@
...
+				->where($db->quoteName('dashboard_id') . ' = ' . (int) $value);
+			$db->setQuery($query);
+
+			try
+			{
+				$title = $db->loadResult();
+			}
+			catch (RuntimeException $e)
+			{
+				JError::raiseWarning(500, $e->getMessage());
```
Comment (thite-amol) [inline src/components/com_tjdashboard/administrator/models/fields/modaldashboard.php]: @monikaatvowel avoid the use of deprecated functions

File: src/components/com_tjdashboard/administrator/models/fields/modaldashboard.php [hunk 10]
```diff
@@ -0,0 +1,164 @@
...
+			$modalTitle     .= ' &#8212; ' . $this->element['label'];
+		}
+
+		$urlSelect = $linkDashboards . '&amp;function=jSelectDashboard_' . $this->id;
+
+		if ($value)
+		{
+			$db    = JFactory::getDbo();
+			$query = $db->getQuery(true)
+				->select($db->quoteName('title'))
```
Comment (thite-amol) [inline src/components/com_tjdashboard/administrator/models/fields/modaldashboard.php]: @monikaatvowel Load this from dashboard library.

File: src/components/com_tjdashboard/administrator/models/dashboard.php (hunk 5 above)
Comment (monikaatvowel) [reply src/components/com_tjdashboard/administrator/models/dashboard.php]: now just using state

File: src/components/com_tjdashboard/administrator/models/fields/modaldashboard.php [hunk 11]
```diff
@@ -0,0 +1,166 @@
...
+		if (isset($this->element['language']))
+		{
+			$linkDashboard   .= '&amp;forcedLanguage=' . $this->element['language'];
+			$modalTitle     .= ' &#8212; ' . $this->element['label'];
+		}
+
+		$urlSelect = $linkDashboards . '&amp;function=jSelectDashboard_' . $this->id;
+
+		if ($value)
+		{
```
Comment (snehal0904) [inline src/components/com_tjdashboard/administrator/models/fields/modaldashboard.php]: @monikaatvowel  Use getInstance

File: src/components/com_tjdashboard/administrator/models/fields/modaldashboard.php [hunk 12]
```diff
@@ -0,0 +1,166 @@
+<?php
+/**
+ * @package    Com_Tjdashboard
+ * @author     Techjoomla <extensions@techjoomla.com>
```
Comment (thite-amol) [inline src/components/com_tjdashboard/administrator/models/fields/modaldashboard.php]: @monikaatvowel Update this with the standard block comment.

**Pull request merged**